LibRangeCheck-3.0/LibRangeCheck-3.0.lua

```diff
@@ -90,9 +90,9 @@ local GetSpellBookItemInfo = _G.GetSpellBookItemInfo or function(index, spellBan
     spellBank = (spellBank == "spell") and Enum.SpellBookSpellBank.Player or Enum.SpellBookSpellBank.Pet;
   end
   local info = C_SpellBook.GetSpellBookItemInfo(index, spellBank)
-  -- we are looking for "Spell" here, as "FutureSpell" and passives are not working with C_Spell.IsSpellInRange
-  if info and not info.isPassive and info.itemType == Enum.SpellBookItemType.Spell then
-    return info.itemType, info.spellID
+  -- we are looking for main spec "Spell" here, as "FutureSpell" and passives are not working with C_Spell.IsSpellInRange
+  if info and not info.isPassive and info.itemType == Enum.SpellBookItemType.Spell and not info.isOffSpec then
+    return info.itemType, info.spellID, info
```

Comment:
So I dislike that we are checking lots of things inside the compability GetSpellBookItemInfo function. Imho ideally, the function should be a as close as possible to the behaviour of classic and retail. 

That is I think, the checking for isOffSpec, isPassive shouldn't be done in the compability function, but in the place where it is used.

Though I'm okay with merging the code as is.

Reply:
I agree and can rework that later today.

Reply:
I updated the code to move the filtering out.